### DISCUSSION TITLE
Suggestion: take a look at PNpm

### DISCUSSION THREAD

**Discussion by Indy-rbo** [vmware/build-tools-for-vmware-aria#396] [Ideas]
### Description

I initially wanted to ask if the dependency on `@type/node` in `vropkg` could be upgraded to at least something like `^14.0.13`. As the older version in `vropkg` currently interferes with some local tooling we have built around the Build Tools. (We currently work around it by upgrading this dependency locally ourselves)

But as I saw some plans for big changes to the project while you are working towards 3.0.0 I thought of a different suggestion to maybe help you manage this repository in the future.

It might be too much work to restructure if you don't see the benefit, but since this repository is already structured like a monorepo I thought now would possibly be a fitting time to suggest maybe switching to [PNpm](https://pnpm.io/).

While PNpm offers greater speeds and doesn't store duplicates of your node modules locally, this would also allow you to use [workspaces](https://pnpm.io/workspaces) and [catalogs](https://pnpm.io/catalogs), which might be of use to you.

### Example

The dependency `@types/node` has different versions within the codebase. 
- vrotsc uses: `^14.0.0`
- `vropkg` uses: `^11.15.7`
- `npmconv` uses: `^12.0.10`

With the help of PNpm [catalogs](https://pnpm.io/catalogs) you are able to specify in your `pnpm-workspace.yaml` which resides in the root:
```
catalog:
  # Can be referenced through "catalog:default" (or "catalog:" as shorthand)
  @types/node: ^22.0.0
  glob: ^11.0.0
  @types/glob: ^11.0.0

catalogs:
  # Can be referenced through "catalog:node14"
  node14:
    @types/node: ^14.0.0

  # Can be referenced through "catalog:node18"
  node18:
    @types/node: ^18.0.0
```

Then within the package specific `package.json` you can use it like:
```
# package.json
  "devDependencies": {
    "@types/node": "catalog:node14",
    "@types/glob": "catalog:",
    "glob": "catalog:"
  }
```

### Alternatives

- Only update the `@types/node` of `vropkg` to `^14.0.0`
- Manually stay up to date with dependencies when possible

This suggestion is mostly to make you aware of the possible option :)

**Comment by VenelinBakalov** [comment]
Closing as discussion as this suggestion was added as part of an open issue: https://github.com/vmware/build-tools-for-vmware-aria/issues/332
